pages/docs/reference/whatsnew14.md

```diff
@@ -6,6 +6,293 @@ title: "What's New in Kotlin 1.4"
+
+```kotlin
+// FILE: test.kt
+fun test(r1: Runnable) {
+    A.foo(r1) {}  // Works now
+}
+```
+
+</div>
+
+#### Java SAM interfaces in Kotlin
+
+Starting from Kotlin 1.4, you can use Java SAM interfaces in Kotlin. Previously, you could declare such a function `foo`
+only in Java code.
+
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+
+```kotlin
+fun foo(r1: java.lang.Runnable) {}
+
+fun test() { 
+    foo { } // OK
+}
+```
+
+</div>
+
+### Callable reference improvements
+
+Kotlin 1.4 supports more cases for using callable references:
+
+* References to functions with default argument values
+* Function references in `Unit`-returning functions 
+* References to functions with a variable number of arguments (`vararg`)
+* Suspend conversion on callable references in Kotlin/JVM
+
+#### References to functions with default argument values 
+
+Using callable references to functions with default argument values is now more convenient. For example, the callable reference 
+to the following `foo` function can be interpreted both as taking one `Int` argument or taking no arguments.
+
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+
+```kotlin
+fun foo(i: Int = 0): String = "$i!"
+
+fun apply1(func: () -> String): String = func()
+fun apply2(func: (Int) -> String): String = func(42)
+
+fun main() {
+    println(apply1(::foo))
+    println(apply2(::foo))
+}
+```
+
+</div>
+
+Previously, you should write additional overloads for the function `apply` to use the default argument values.
+
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+
+```kotlin
+// some new overload
+fun applyInt(func: (Int) -> String): String = func() 
+```
+
+</div>
+
+#### Function references in `Unit`-returning functions 
+
+Now you can use callable references to functions returning any type in `Unit`-returning functions. The Unit-returning 
+function `foo()` uses a callable reference to `returnsInt()` returning `Int` as a result.
+
+<div class="sample" markdown="1" theme="idea" data-min-compiler-version="1.4">
+
+```kotlin
+fun foo(f: () -> Unit) { }
+fun returnsInt(): Int = 42
+
+fun main() {
+    println(foo(::returnsInt))
+}
+```
+
+</div>
+
+#### References to functions with a variable number of arguments
+
+Starting from Kotlin 1.4, you can use callable references to functions with a variable number of arguments (`vararg`) 
+at the end. You can use any number of arguments of the same type at the end.
+
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+
+```kotlin
+fun foo(x: Int, vararg y: String) {}
+
+fun use0(f: (Int) -> Unit) {}
+fun use1(f: (Int, String) -> Unit) {}
+fun use2(f: (Int, String, String) -> Unit) {}
+
+fun test() {
+    use0(::foo) 
+    use1(::foo) 
+    use2(::foo) 
+}
+```
+
+</div>
+
+#### Suspend conversion on callable references
+
+Kotlin 1.4 supports suspend conversion on callable references in Kotlin/JVM. 
```

Comment:
Same as with callable references, I think we can add that it worked with lambdas and now with it works with callable references as well:
```
fun call() {}
fun takeSuspend(f: suspend () -> Unit) {}
fun test() {
    takeSuspend { call() } // OK before 1.4
    takeSuspend(::call) // Now this case also works
}
```

Reply:
✅